### PR TITLE
[VL] RAS: Avoid adding R2C whose schema contains complex data types

### PR DESCRIPTION
**Pull request by zhztheplayer**
It's observed Velox R2C becomes expensive when containing complex data types as input. Add a coster in rough cost model to address the issue.

To enable rough cost model:

```
spark.gluten.ras.enabled=true
spark.gluten.ras.costModel=rough
```

### CODE REVIEW

**Comment by zhztheplayer** [comment]
cc @FelixYBW @jinchengchenghh

File: gluten-substrait/src/main/scala/org/apache/gluten/planner/cost/RoughCostModel.scala [hunk 1]
```diff
@@ -34,6 +36,10 @@ class RoughCostModel extends LongCostModel {
         // Make trivial ProjectExec has the same cost as ProjectExecTransform to reduce unnecessary
         // c2r and r2c.
         10L
+      case r2c: RowToColumnarExecBase if hasComplexTypes(r2c.schema) =>
+        // Avoid moving computation back to native when transition has complex types in schema.
+        // Such transitions are observed to be extreme expensive as of now.
+        Long.MaxValue
```
Comment (jinchengchenghh) [inline gluten-substrait/src/main/scala/org/apache/gluten/planner/cost/RoughCostModel.scala:42]: Can we add a more reasonable value to the cost? How can we measure the operator cost now?

Comment (jinchengchenghh) [reply gluten-substrait/src/main/scala/org/apache/gluten/planner/cost/RoughCostModel.scala:45]: We can use it as a temp solution. The r2c is very expensive.

**Review by jinchengchenghh — DISMISSED** [review]

File: gluten-substrait/src/main/scala/org/apache/gluten/planner/cost/RoughCostModel.scala (hunk 1 above)
Comment (zhztheplayer) [reply gluten-substrait/src/main/scala/org/apache/gluten/planner/cost/RoughCostModel.scala:43]: Yes. It's just a heuristic as a temp solution.

**Comment by FelixYBW** [comment]
temp fix of #7223

**Review by JkSelf — APPROVED** [review]

**Comment by FelixYBW** [comment]
Looks it doesn't work. Let me check what happens

**Pull request merged**